src/h5web/visualizations/nexus/utils.ts

```diff
@@ -124,15 +125,49 @@ export function getLinkedDatasets(
   return Object.fromEntries(datasetEntries);
 }
 
-export function getNxAxesParams(
-  nxDatasets: Record<string, HDF5Dataset>,
-  datasetValues: Record<string, HDF5Value> | undefined
-): Record<string, AxisParams> {
-  const paramsEntries = Object.entries(nxDatasets).map(([axis, dataset]) => {
-    const values = datasetValues && datasetValues[axis];
-    assertOptionalArray<number>(values);
-    return [axis, { values, label: getDatasetLabel(dataset, axis) }];
+export function useParsedNxData(
+  group: HDF5Group,
+  metadata: HDF5Metadata
+): NxData | undefined {
+  const datasetValues = useChildDatasetValues(group, metadata);
+  if (!datasetValues) {
+    return undefined;
+  }
+
+  const signalName = getAttributeValue(group, 'signal');
+  assertStr(signalName);
+  const signalDataset = getLinkedEntity(signalName, group, metadata);
+  const signalValue = datasetValues[signalName];
+
+  if (!signalDataset || !isDataset(signalDataset) || !signalValue) {
+    return undefined;
+  }
+  assertArray<number>(signalValue);
+
+  const signal = {
+    label: getDatasetLabel(signalDataset, signalName),
+    value: signalValue,
+    dims: (signalDataset.shape as HDF5SimpleShape).dims,
+  };
+
+  const axisMapping = getNxAxisMapping(group);
+  const axisDatasets = getLinkedDatasets(
+    axisMapping.filter((v): v is string => !!v),
+    group,
+    metadata
+  );
+
+  const axesEntries = Object.entries(axisDatasets).map(([name, dataset]) => {
+    const value = datasetValues[name];
+    assertOptionalArray<number>(value);
+    return [name, { value, label: getDatasetLabel(dataset, name) }];
   });
```

Comment:
When you end up refactoring `axisMapping`+`axesParams` into a single `axisParamsMapping`(?) array, this bit will become much simpler: you'll just have to map through `axisMapping`, and for each value, call `getLinkedEntity`, assert the entity and the value, and return an `AxisParams` object.